### PR TITLE
Add default unit

### PR DESCRIPTION
**Pull request by maykremer**

### CODE REVIEW

File: src/pages/UserManagement/components/UnitDetails/UnitDetails.jsx [hunk 1]
```diff
@@ -6,18 +6,27 @@ import DashboardCard from '../../../../common/DashboardCard/DashboardCard.jsx';
 import UnitDetailsHeadLine from '../UnitDetailsHeadLine/UnitDetailsHeadLine';
 import UnitDetailsUsersTable from '../UnitDetailsUsersTable/UnitDetailsUsersTable';
 import UserService from '../../../../services/user.service.js';
+import config from '../../config';
 import NoObjectsToShow from '../../../../common/NoObjectsToShow/NoObjectsToShow';
 
-const UnitDetails = ({ unit, users, setUsers }) => {
+const UnitDetails = ({ unit, users, setUsers, isUserAdmin }) => {
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/UnitDetails/UnitDetails.jsx:12]: remove `isUserAdmin`

File: src/pages/UserManagement/components/UnitDetails/UnitDetails.jsx [hunk 2]
```diff
@@ -26,7 +35,12 @@ const UnitDetails = ({ unit, users, setUsers }) => {
         { unit ? (
           <>
             <UnitDetailsHeadLine unitName={unit.name} numberOfUnitUsers={users.length} />
-            <UnitDetailsUsersTable users={users} setUsers={setUsers} unit={unit} />
+            <UnitDetailsUsersTable
+              users={users}
+              setUsers={setUsers}
+              unit={unit}
+              isUserAdmin={isUserAdmin}
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/UnitDetails/UnitDetails.jsx:42]: remove `isUserAdmin`

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — APPROVED** [review]

File: mockserver/user/user.manager.js [hunk 3]
```diff
@@ -15,6 +15,10 @@ class UserManager {
     res.send(users.find((user) => user.id === req.params.id));
   }
 
+  static async getUsersByRoles(req, res) {
+    res.send(users.filter((user) => req.body.roles.includes(user.role)));
+  }
+
```
Comment (lustigy96) [inline mockserver/user/user.manager.js:21]: add this function to the API

File: src/pages/UserManagement/config.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Role } from '../../services/user.service';
+
+const config = {
+  adminRoles: [Role.Mada, Role.Technical, Role.Itur],
+  superUnitId: '507f191e810c19729de860ea',
```
Comment (lustigy96) [inline src/pages/UserManagement/config.js:5]: why is this the super Id? maybe add remarks toexplain it

File: src/services/user.service.js [hunk 5]
```diff
@@ -28,8 +28,12 @@ export default class UserService {
     const res = await axios.get(`${await config.apiUri}/api/user/`, { headers, params });
     return res?.data;
   }
-  static async createUser(unitId, role, name) {
-    const res = await axios.post(`${await config.apiUri}/api/user/`, { unitId, role, name }, { headers });
+  static async createUser(userToAdd) {
+    const res = await axios.post(`${await config.apiUri}/api/user/`, userToAdd, { headers });
+    return res?.data;
+  }
```
Comment (lustigy96) [inline src/services/user.service.js:34]:
```suggestion
  static async createUser(userToAdd) {
    const { data }= await axios.post(`${await config.apiUri}/api/user/`, userToAdd, { headers });
    return data ;
  }
```

File: src/services/user.service.js [hunk 6]
```diff
@@ -28,8 +28,12 @@ export default class UserService {
...
+  static async getUsersByRoles(roles) {
+    const res = await axios.post(`${await config.apiUri}/api/user/role`, { roles }, { headers });
     return res?.data;
   }
```
Comment (lustigy96) [inline src/services/user.service.js:38]:
```suggestion
    const { data }= await axios.post(`${await config.apiUri}/api/user/role`, { roles }, { headers });
    return data;
  }
```

File: src/services/user.service.js [hunk 7]
```diff
@@ -28,8 +28,12 @@ export default class UserService {
     const res = await axios.get(`${await config.apiUri}/api/user/`, { headers, params });
     return res?.data;
   }
-  static async createUser(unitId, role, name) {
-    const res = await axios.post(`${await config.apiUri}/api/user/`, { unitId, role, name }, { headers });
+  static async createUser(userToAdd) {
+    const res = await axios.post(`${await config.apiUri}/api/user/`, userToAdd, { headers });
+    return res?.data;
```
Comment (lustigy96) [inline src/services/user.service.js:33]:
```suggestion
    const { data }= await axios.post(`${await config.apiUri}/api/user/`, userToAdd, { headers });
    return data;
```

**Review by lustigy96 — CHANGES_REQUESTED** [review]

**Pull request merged**